jupyter-notebooks/data-api-tutorials/planet_python_client_introduction.ipynb

```diff
@@ -24,7 +24,7 @@
     "\n",
     "## Useful links \n",
     "* [Documentation](https://planetlabs.github.io/planet-client-python/index.html)\n",
-    "* [Planet Data API reference](https://www.planet.com/docs/reference/data-api/)"
+    "* [Planet Data API reference](https://developers.planet.com/docs/apis/data/)"
```

Comment:
This comment is buried below, but for the whole notebook, and all notebooks, be sure to run all of the cells before `git commit` so that the output can be stored. The ability for a reader to view a rendered notebook on github and see the output along with the code is a big reason we use notebooks.

## Requirements

Maybe specify that the version of `planet` is 2

## Set up

`p` is not a great name for a function because it is easy to forgot what it does. I'd consider removing it entirely and replacing it with the one-liner. Maybe use a global var for `INDENT` so it's `print(json.dumps(data, indent=INDENT))`

```python
# if your Planet API Key is not set as an environment variable, you can paste it below
if os.environ.get('PL_API_KEY', ''):
    API_KEY = os.environ.get('PL_API_KEY', '')
else:
    API_KEY = 'PASTE_YOUR_API_KEY_HERE'

client = Auth.from_key(API_KEY)

# Setup the session
session = requests.Session()

# Authenticate
session.auth = (API_KEY, "")
```
I'm surprised we are using requests here. And I don't think the auth stuff is required anymore if we just tell them to use an environment variable.

```python
geom_filter = data_filter.geometry_filter(geom)
clear_percent_filter = data_filter.range_filter('clear_percent', 90)
date_range_filter = data_filter.date_range_filter("acquired", datetime(month=1, day=1, year=2017))
cloud_cover_filter = data_filter.range_filter('cloud_cover', None, 0.1)
```
use keys for kwargs, not positions (see previous comment for an example)

```python
p(combined_filter)
```
I think maybe the idea here was to store the output so that someone looking at the rendered notebook can follow along. So for this notebook and for the others you upload, make sure that all the cells are run and the output is there for the user to read e.g. on github.

```python
async with Session() as sess:
    cl = DataClient(sess)
    request = await cl.create_search(name='planet_client_demo',search_filter=combined_filter, item_types=item_types)
```
no need to specify these as keyword arguments, they are positional (aka have no default values)
```python
# Search the Data API
async with Session() as sess:
    cl = DataClient(sess)
    items = await cl.run_search(search_id=request['id'])
    item_list = [i async for i in items]
```
same comment re: positional args

## Assets and downloads
Here, we can use the new asset functions in the DataClient, e.g. `get_asset`, `activate_asset` and `wait_asset`